javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/concurrent/InstrumentedTaskClasses.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.api.concurrent;
+
+import io.opentelemetry.instrumentation.api.config.Config;
+import io.opentelemetry.javaagent.instrumentation.api.util.Trie;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class InstrumentedTaskClasses {
+
+  private static final Logger log = LoggerFactory.getLogger(Config.class);
+
+  private static final String AGENT_CLASSLOADER_NAME =
+      "io.opentelemetry.javaagent.bootstrap.AgentClassLoader";
+
+  private static final ClassValue<Boolean> INSTRUMENTED_TASK_CLASS =
+      new ClassValue<Boolean>() {
+        @Override
+        protected Boolean computeValue(Class<?> taskClass) {
+          // do not instrument ignored task classes
+          if (ignoredTaskClasses.getOrDefault(taskClass.getName(), false)) {
+            return false;
+          }
+          // Don't trace runnables from libraries that are packaged inside the agent.
+          // Although GlobalIgnoredTypesConfigurer excludes these classes from instrumentation
+          // their instances can still be passed to executors which we have instrumented so we need
+          // to exclude them here too.
+          ClassLoader taskClassLoader = taskClass.getClassLoader();
+          if (taskClassLoader != null
+              && AGENT_CLASSLOADER_NAME.equals(taskClassLoader.getClass().getName())) {
+            return false;
+          }
+          return true;
+        }
+      };
+
+  private static volatile Trie<Boolean> ignoredTaskClasses;
+
+  /**
+   * Sets the configured ignored tasks trie. This method is called internally from the agent
```

Comment:
Isn't this whole class / mechanism basically internal?

Reply:
Hmm, that's true, I guess that it's very similar to `BootstrapPackagePrefixesHolder` in that regard. I'll move it to the `internal` package. Thanks!